### DISCUSSION TITLE
Treating missing values

### DISCUSSION THREAD

**Discussion by igalriklin** [vdemichev/DiaNN#956] [General]
Hi Vadim,

I ran 2 samples from the same HELA cells, and compared the results. Then I ran only these 2 samples in DIANN. I log transformed all the values, and treated the missing values as zeros.
Here is an image comparing the values of each sample.
![image](https://github.com/vdemichev/DiaNN/assets/44479769/4708b854-f6a4-4d8f-b530-bacaa7b1f136)
I expected that the zero values for sample will be very low in the other sample, close to the detection limit. However when I made the histogram of the values of each sample when the other are zeros, I was surprised to find out that they spread on almost the whole range of values.
![image](https://github.com/vdemichev/DiaNN/assets/44479769/163c3a16-2a5b-4ebd-8bd7-17d3d9ea2746)
Does this looks like an error to you? I think that it means that I cannot use imputation with zero or with minimal values for the unknown measurments. Do you agree? How is the best they to treat them in your opinion?

Thanks!
Igal

**Comment by vdemichev** [comment]
Hi Igal,

This is normal & expected.
Also, if you use MBR, the missing value rate will go down.

We use minimal value imputation (when we do that - often we don't impute at all) like this:
- log transform the quantities and create a matrix with rows - proteins and columns - samples
- for each row, determine the lowest valid number in that row and replace all NAs in that row with this number

Best,
Vadim
